src/android/com/microsoftopentechnologies/intellij/helpers/azure/sdk/AzureSDKManagerImpl.java

```diff
@@ -380,73 +368,457 @@ public void deleteVirtualMachine(@NotNull VirtualMachine vm, boolean deleteFromS
+            StorageAccountCreateParameters sacp = new StorageAccountCreateParameters(storageAccount.getName(),
+                    storageAccount.getName());
+            sacp.setAccountType(storageAccount.getType());
+            if (!storageAccount.getAffinityGroup().isEmpty()) {
+                sacp.setAffinityGroup(storageAccount.getAffinityGroup());
+            } else if (!storageAccount.getLocation().isEmpty()) {
+                sacp.setLocation(storageAccount.getLocation());
+            }
+
+            OperationStatusResponse osr = sao.create(sacp);
+            validateOperationStatus(osr);
+        } catch (Throwable t) {
+            throw new AzureCmdException("Error creating the Storage Account", t);
+        } finally {
+            if (client != null) {
+                try {
+                    client.close();
+                } catch (IOException ignored) {
+                }
+            }
+        }
+    }
+
+    @Override
+    public void createCloudService(@NotNull CloudService cloudService) throws AzureCmdException {
+        ComputeManagementClient client = null;
+
+        try {
+            client = getComputeManagementClient(cloudService.getSubscriptionId());
+            HostedServiceOperations hso = getHostedServiceOperations(client);
+            HostedServiceCreateParameters hscp = new HostedServiceCreateParameters(cloudService.getName(),
+                    cloudService.getName());
+
+            if (!cloudService.getAffinityGroup().isEmpty()) {
+                hscp.setAffinityGroup(cloudService.getAffinityGroup());
+            } else if (!cloudService.getLocation().isEmpty()) {
+                hscp.setLocation(cloudService.getLocation());
+            }
+
+            OperationResponse or = hso.create(hscp);
+
+            if (or == null) {
+                throw new Exception("Unable to retrieve Operation");
+            }
+
+            OperationStatusResponse osr = getOperationStatusResponse(client, or);
+            validateOperationStatus(osr);
+        } catch (Throwable t) {
+            throw new AzureCmdException("Error creating the Cloud Service", t);
+        } finally {
+            if (client != null) {
+                try {
+                    client.close();
+                } catch (IOException ignored) {
+                }
+            }
+        }
+    }
+
+    @Override
+    public void createVirtualMachine(@NotNull VirtualMachine virtualMachine, @NotNull VirtualMachineImage vmImage,
+                                     @NotNull StorageAccount storageAccount, @NotNull String username,
+                                     @NotNull String password)
+            throws AzureCmdException {
+        try {
+            String mediaLocation = getMediaLocation(virtualMachine, storageAccount);
+
+            createVirtualMachine(virtualMachine, vmImage, mediaLocation, username, password);
+        } catch (AzureCmdException e) {
+            throw e;
```

Comment:
If the exception is already an AzureCmdException it will bubble up. For any other exception (catch Throwable below) we wrap it in an AzureCmdException.